### DISCUSSION TITLE
Crash when running application built with OTP 24.0

### DISCUSSION THREAD

**Discussion by ghost** [rabbitmq/credentials-obfuscation#14] [General]
Hello.
When I'm running my application used credentials_obfuscation built with OTP 24.0, I got the following error:
`
16:48:25.658 [error] Supervisor credentials_obfuscation_sup had child credentials_obfuscaton_svc started with credentials_obfuscation_svc:start_link() at undefined exit with reason call to undefined function crypto:hmac(sha256, <<81,199,27,16,7,18,221,254,2,216,92,143,103,242,11,83,157,56,144,201,125,68,144,87,110,238,253,...>>, [<<247,134,149,166,201,240,114,137,102,151,74,75,201,76,117,199>>,<<0,0,0,1>>], 32) in context start_error
`
In OTP 24.0 the function crypto:hmac was replaced with crypto:mac*: https://github.com/erlang/otp/blob/3a8e65972ac1d0c4ae1412eae2dc2b3e5b3d758d/lib/crypto/src/crypto.erl#L54

**Comment by michaelklishin** [comment]
This library [does not use `crypto:hmac/4`](https://github.com/rabbitmq/credentials-obfuscation/search?q=crypto%3Ahmac) and is a part of RabbitMQ which is Erlang 24-compatible.

I cannot suggest much without an executable way to reproduce.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
https://github.com/rabbitmq/credentials-obfuscation/pull/10 introduced Erlang 24 support a couple of months ago. Here is where it [replaced a call site of `crypto:hmac/4`](https://github.com/rabbitmq/credentials-obfuscation/pull/10/files#diff-94a2f471028bbf0ecbcaaf91a1e8ea7ebbe1ccb704380464484ef44bc7ca8f44L126). You may want to upgrade to the latest release.

In the future, consider submitting PRs for problems you understand the root cause of. This is open source software after all.

**Comment by ghost** [comment]
We used amqp_client version 3.8.11. After updating to 3.8.14 all works. Thanks.
